src/app/ubs/ubs-admin/components/ubs-admin-order-details-form/ubs-admin-order-details-form.component.ts

```diff
@@ -198,14 +216,17 @@ export class UbsAdminOrderDetailsFormComponent implements OnInit, OnChanges {
 
   private calculateOverpayment() {
     const bagType = this.orderStatusInfo.ableActualChange ? 'actual' : 'confirmed';
-    let priceWithoutCertificate = this.bagsInfo.sum[bagType] - this.orderDetails.certificateDiscount;
-
+    let priceWithoutCertificate = this.bagsInfo?.sum[bagType] - this.orderDetails.certificateDiscount;
     priceWithoutCertificate = Math.max(priceWithoutCertificate, 0);
 
-    this.overpayment = this.orderDetails.bonuses + this.orderDetails.paidAmount - priceWithoutCertificate - this.writeoffAtStationSum;
+    const baseSumOfOrder = this.orderDetails.bonuses + this.orderDetails.paidAmount + this.orderDetails.certificateDiscount;
 
-    if (this.showUbsCourier) {
-      this.overpayment += this.courierPrice * (this.isOrderCancelled ? -1 : 1);
+    if (this.isOrderBroughtByHimself) {
+      this.overpayment = baseSumOfOrder - this.writeoffAtStationSum;
+    } else if (this.showUbsCourier) {
```

Comment:
why you use 'else if' here?

Reply:
'else if' in this case allows to handle multiple scenarios for calculating the overpayment based on [different conditions](https://github.com/ita-social-projects/GreenCity/issues/5520).